File: stack.cpp
#include <iostream>
using namespace std;

int stack[10];
int size = 10;
int top = -1;

bool isEmpty(){
    if(top == -1){
        return true;
    }
    return false;
}
bool isFull(){
    if(top == size){
        return true;
    }
    return false;
}
void push(int data){
    if(isFull()){
        cout <<"Stack Overflow"<<endl;
    }else{
        top++;
        stack[top] = data;
    }
}
int pop(){
    if(isEmpty()){
        cout <<"Stack Underflow"<<endl;
        return -1;
    }        
    return stack[top--];
}
void topView(){
    if(isEmpty()){
        cout <<"Stack Underflow"<<endl;
    }else{
        cout << stack[top];
    }
    
}

void printStack(){
    if(isEmpty()){
        cout <<"Stack Underflow"<<endl;
    }else{
        while(!isEmpty()){
            cout <<pop()<<" - ";
        }
    }
}

int main() {
    
    push(10);
    push(25);
    push(3);
    pop();
    push(5);
    push(56);
    push(55);
    push(10);
    push(8);
    push(96);
    push(81);
    push(02);
    // cout<<pop();
    printStack();
     

}

File: stack.java
public class Main {
    public static void main(String args[]) {
        Stack s = new Stack();
        for(int i = 0; i<10; i++){
            s.push(i*i);
        }
        // System.out.println(s.peek());
        while(!s.isEmpty()){
            System.out.print(s.st[s.top--] + " - ");
        }
        
    }
}

class Stack{
    int st[] = new int[10];
    int top = -1;

    boolean isEmpty(){
        if(top == -1){
            return true;
        }
        return false;
    }
    boolean isFull(){
        if(top == st.length){
            return true;
        }
        return false;
    }
    void push(int data){
        if(isFull()){
            System.out.println("Stack Overflow");
        }else{
            st[++top] = data;
        }
    }

    int pop(){
        if(isEmpty()){
            System.out.println("Stack underflow");
            return -1;
        }
        // int val = stack[top--];
        return st[top--];
    }
    int peek(){
        if(isEmpty()){
            System.out.println("Stack underflow");
            return -1;
        }
        return st[top];
    }

}